xsdvalidate/xsd_validate_activity.go

```diff
@@ -0,0 +1,192 @@
+package xsdvalidate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	xsdvalidator "github.com/terminalstatic/go-xsd-validate"
+	"go.artefactual.dev/tools/temporal"
+)
+
+const XSDValidateActivityName = "xsd-validate-activity"
+
+type XSDValidateActivity struct{}
+
+func NewXSDValidateActivity() *XSDValidateActivity {
+	return &XSDValidateActivity{}
+}
+
+type XSDValidateActivityParams struct {
+	// DirectoryPath is the path of the files to be validated.
+	DirectoryPath string
+
+	// Pattern is the regular expression for files to validate.
+	Pattern string
+
+	// XSDFilePathOrURL is the path of the XSD file to use for validation.
+	XSDFilePathOrURL string
+}
+
+type XSDValidateActivityResult struct {
+	Failed []string
+}
+
+// Execute checks all files, in a given directory, matching a certain pattern.
+func (a *XSDValidateActivity) Execute(
+	ctx context.Context,
+	params *XSDValidateActivityParams,
+) (*XSDValidateActivityResult, error) {
+	logger := temporal.GetLogger(ctx)
+	logger.V(1).Info("Executing XMLValidateActivity", "DirectoryPath", params.DirectoryPath)
+
+	// Check is XSD has been provided as a file, otherwise attempt download.
+	_, err := os.Stat(params.XSDFilePathOrURL)
+	if errors.Is(err, os.ErrNotExist) {
+		// Attempt to download XSD file.
+		file, err := os.CreateTemp("", "xsdValidatrActvitity")
+		if err != nil {
+			return nil, err
+		}
+		defer os.Remove(file.Name())
+
+		err = DownloadXsdFile(params.XSDFilePathOrURL, file.Name())
```

Comment:
@sevein has drilled into me "avoid exporting functions (and variables, etc.) unless you need to call them from outside of the package". My default was to export symbols "just in case" but @sevein has convinced me (after much repetition) that it's better to keep the surface of the package as small as possible to make it easier to update without breaking compatibility.

Reply:
Fixed!